### PR TITLE
Feature/core name

### PR DESCRIPTION
**Pull request by sitepark-veltrup**
Extensions for

- Automatic determination of the index name
- Extensions for date filters and facets

### CODE REVIEW

File: src/Service/EnvVarLoader.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    private function determineResourceRoot(): string
+    {
+        /** @var string[] $directories */
+        $directories = [
+            $this->baseDir
+        ];
+
+        if (isset($_SERVER['SCRIPT_FILENAME'])) {
+            $binDir = dirname($_SERVER['SCRIPT_FILENAME']);
```
Comment (sitepark-schaeper) [inline src/Service/EnvVarLoader.php:69]:
```suggestion
        $filename = $_SERVER['SCRIPT_FILENAME'] ?? null;
        if (is_string($filename)) {
            $binDir = dirname($filename);
```

File: src/Service/EnvVarLoader.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            }
+            if (is_file($realpath . '/context.php')) {
+                return $realpath;
+            }
+            if (is_file($realpath . '/WEB-IES/context.php')) {
+                return $realpath;
+            }
+        }
+
+        return '';
```
Comment (sitepark-schaeper) [inline src/Service/EnvVarLoader.php:92]: why does this return an empty string on failure?

File: src/Service/EnvVarLoader.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    ) {
+        $this->baseDir = $baseDir ?? (getcwd() ?: '');
+    }
+
+    public function loadEnvVars(): array
+    {
+        $env = [];
+
+        $resourceRoot = $_SERVER['RESOURCE_ROOT'] ?? '';
+        if (empty($resourceRoot)) {
```
Comment (sitepark-schaeper) [inline src/Service/EnvVarLoader.php:27]:
```suggestion
        if (!is_string($resourceRoot) || empty($resourceRoot)) {
```

File: src/Service/EnvVarLoader.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        foreach ($directories as $dir) {
+            if (is_file($dir . '/.env')) {
+                $dotenv = new Dotenv();
+                $dotenv->load($dir . '/.env');
+                return 'https://solr-' . $_ENV['SERVER_BASE_NAME'];
+            }
+        }
+
+        return '';
```
Comment (sitepark-schaeper) [inline src/Service/EnvVarLoader.php:117]: why does this return an empty string on failure?

File: src/Service/EnvVarLoader.php [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+     * the .env file of the Docker environment. This contains the
+     * SERVER_BASE_NAME variable, which can be used to determine the SOLR_URL.
+     */
+    private function determineSolrUrl(string $resourceRoot): string
+    {
+
+        $directories = [
+            $resourceRoot . '/../../../..',
+            $resourceRoot . '/../../../../..',
+        ];
```
Comment (sitepark-schaeper) [inline src/Service/EnvVarLoader.php:107]: this needs an explaination / a better implementation.

File: src/Service/EnvVarLoader.php [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+    public function loadEnvVars(): array
+    {
+        $env = [];
+
+        $resourceRoot = $_SERVER['RESOURCE_ROOT'] ?? '';
+        if (empty($resourceRoot)) {
+            $resourceRoot = $this->determineResourceRoot();
+            if (!empty($resourceRoot)) {
+                $env['RESOURCE_ROOT'] = $resourceRoot;
```
[the remark this answers is not in the document]
Comment (sitepark-veltrup) [reply src/Service/EnvVarLoader.php:30]: Yes, if it is already set in `$_SERVER` it goes its usual way and does not have to be set again via the `EnvVarLoader`.

File: src/Service/EnvVarLoader.php (hunk 4 above)
Comment (sitepark-veltrup) [reply src/Service/EnvVarLoader.php:129]: It is not really an error. Here the Solr URL can only be determined if the call is made via the console and the current directory is a WWW directory in an IES environment. This is only the case on development systems. This is only a tool for development systems. The Solr URL can never be determined using this on production systems.

File: src/Service/EnvVarLoader.php (hunk 5 above)
Comment (sitepark-veltrup) [reply src/Service/EnvVarLoader.php:119]: I'll add a few comments here.

File: src/Input/RelativeDateRangeInputFilter.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    #[GQL\Field(type:"DateInterval")]
+    public ?\DateInterval $before = null;
+
+    #[GQL\Field(type:"DateInterval")]
+    public ?\DateInterval $after = null;
+    #[GQL\Field(type:"DateRangeRound")]
+    public ?DateRangeRound $roundStart = null;
+    #[GQL\Field(type:"DateRangeRound")]
+    public ?DateRangeRound $roundEnd = null;
+}
```
[the remark this answers is not in the document]
Comment (sitepark-veltrup) [reply src/Input/RelativeDateRangeInputFilter.php]: Is corrected. too bad phpcs can't find it.

File: src/Input/RelativeDateRangeInputFacet.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    #[GQL\Field(type:"DateInterval")]
+    public ?\DateInterval $after = null;
+
+    #[GQL\Field(type:"DateInterval")]
+    public ?\DateInterval $gap = null;
+    #[GQL\Field(type:"DateRangeRound")]
+    public ?DateRangeRound $roundStart = null;
+    #[GQL\Field(type:"DateRangeRound")]
+    public ?DateRangeRound $roundEnd = null;
+}
```
[the remark this answers is not in the document]
Comment (sitepark-veltrup) [reply src/Input/RelativeDateRangeInputFacet.php]: Is corrected. Too bad phpcs can't find it.

File: src/Resolver/NewsTeaserFactory.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        $url = $this->urlRewriter->rewrite(
+            UrlRewriterType::LINK,
+            $resource->location
+        );
+
+        $headline = $resource->data->getString(
+            'base.teaser.headline',
+            $resource->name
+        );
+        $text = $resource->data->getString('base.teaser.text');
```
[the remark this answers is not in the document]
Comment (sitepark-schaeper) [reply src/Resolver/NewsTeaserFactory.php:29]: hm, so the nullable `NewsTeaser::$text` will not actually be `null` but `''`? that does not sound ideal if I'm beeing honest.

**Review by sitepark-schaeper — COMMENTED** [review]
soley [the comment from my last review](https://github.com/sitepark/atoolo-graphql-search-bundle/pull/43#pullrequestreview-2049908936) is now unaddressed

**Comment by sitepark-veltrup** [comment]
> soley [the comment from my last review](https://github.com/sitepark/atoolo-graphql-search-bundle/pull/43#pullrequestreview-2049908936) is now unaddressed

Oh, I overlooked that. The GitHub actions seem to have a problem right now. But the project can be validated properly.

**Review by sitepark-schaeper — APPROVED** [review]
alright, looks good :+1: 

you may want to look into the one thing I found, but I am aware that it currently has no effect.

**Pull request merged**